src/features/items.ts

```diff
@@ -0,0 +1,134 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { ListItem } from "../types/ListItem";
+import { Tag } from "../types/Tag";
+
+export const ADD_NEW_ITEM = "ADD_NEW_ITEM";
+export type AddNewItemActionType = {
+  type: typeof ADD_NEW_ITEM;
+  payload: ListItem;
+};
+export const addNewItem: (url: string) => AddNewItemActionType = (url) => {
+  return {
+    type: ADD_NEW_ITEM,
+    payload: {
+      id: (ItemsStore.getState().length + 1).toString(),
+      url: url,
+      date: new Date().getTime().toString(),
+      alreadyRead: false,
+      tags: [],
+    },
+  };
+};
+
+export const ASSIGN_TAG_TO_ITEM = "ASSIGN_TAG_TO_ITEM";
+export type AssignTagToItemActionType = {
+  type: typeof ASSIGN_TAG_TO_ITEM;
+  payload: ListItem;
+};
+export const assignTagToItem: (
+  tag: Tag,
+  item: ListItem
+) => AssignTagToItemActionType = (tag, item) => {
+  return {
+    type: ASSIGN_TAG_TO_ITEM,
+    payload: {
+      ...item,
+      tags: [...item.tags, tag],
+    },
+  };
+};
+
+export const UNASSIGN_TAG_FROM_ITEM = "UNASSIGN_TAG_FROM_ITEM";
+export type UnassignTagFromItemActionType = {
+  type: typeof UNASSIGN_TAG_FROM_ITEM;
+  payload: { tag: Tag; item: ListItem };
```

Comment:
Again, would get rid of `payload` and rather have two properties `tag` & `item` defined in this type. No need for one more level in hierarchy.